sassanh/qnvim
# feat: new cmdline (local to editor view)

## Summary

This creates Neovim command line inside the editor, instead of occupying Qt Creator search bar. This is an enhancement, because:
1. This makes it more prominent and easier to notice (the cmd lines spawns inside the split we are currently viewing).
2. We could write longer lines for commands.
3. We could now see full messages Neovim prints. If there is not enough place, we can scroll the message.

## Notes

- I found some strange behavior when closing splits, but it existed before my change.
- When you start typing the text in the command line, but then change focus using mouse to editor, your cursor is forced to be in the command line, but the editor highlights behave strangely. I am not sure how this behavior should be handled, because in Neovim, when you click the mouse, nothing really changes: you could only select text in the splits, but command line stays active. Should we do the same thing, or should we discard the command line and bring the focus to the editor, that was clicked on?

## Test Plan

1. Press `:`, `/` or `?` and try to use some neovim command. It should work.
2. Try to move cursor inside command line. It should work.
3. Try to print message in the command line, e.g. using an incorrect command.
4. Try to print multiline message in the command line using `:history` or `echo "something\n\n\n\n\nsomething"`.
5. Find other ways to break the command line.

Closes #56
